### ISSUE TITLE
MaxAttemptsExceededException during Kafka Connect creation

### ISSUE THREAD

**Issue by serrss** [strimzi/strimzi#257]
### Step to reproduce:
1. Clone and open Strimzi sources https://github.com/strimzi/strimzi
2. Up cluster
`$ oc clucter up`
3. Login as admin
`$ oc login -u system:admin`
4. Deploy Cluster Controller
`$ oc create -f resources/openshift/cluster-controller-with-template.yaml`
5. Deploy Topic Controller
`$ oc create -f resources/openshift/topic-controller-with-template.yaml`
6. Login to OpenShift https://127.0.0.1:8443
7. Create **"Apache Kafka (Persistent storage)"**
7.1 Click "Add to project"
7.2 Click "Select from project"
7.3 Select "Apache Kafka (Persistent storage)" and click "Next"
7.4 Use default configurations:
_Number of Zookeper cluster nodes: 3
Number of Kafka cluster nodes: 3_
7.5 Click "Create" and wait for the message "Apache Kafka (Persistent storage) has been created"
7.6 Open page "Pods" and wait when all pods will have status "Running"
8. Create **"Strimzi Topic Controller"**
8.1 Click "Add to project"
8.2 Click "Select from project"
8.3 Select "Strimzi Topic Controller" and click "Next"
8.4 Use default configurations
8.5 Click "Create" and wait for the message "Success Strimzi Topic Controller has been created."
8.6 Open page "Pods" and wait when all pods will have status "Running"
9. Create **"Apache Kafka Connect"**
9.1 Click "Add to project"
9.2 Click "Select from project"
9.3 Select "Apache Kafka Connect" and click "Next"
9.4 Use default configurations
9.5 Click "Create" and wait for the message "Apache Kafka Connect has been created."
9.6 Open page "Pods" and wait when all pods will have status "Running"
10. Open logs for _strimzi-topic-controller_ and wait 2-3 mins

### Actual result:
Log has 
```
| [2018-02-25 20:17:44,774] WARN  <opicsWatcher:101> [oop-thread-7] Error responding to creation of topic connect-cluster-offsets
  | io.strimzi.controller.topic.MaxAttemptsExceededException
  | at io.strimzi.controller.topic.BackOff.delay(BackOff.java:44)
  | at io.strimzi.controller.topic.BackOff.delayMs(BackOff.java:36)
```

### Additional information:
Openshift - v3.7.1+a8deba5-34
Strimzi - 141 build
All log file will be attached to Jira issue

### Topic controller log: 
```
[2018-02-25 20:16:57,280] INFO  <Main        :42> [oop-thread-4] Session deployed
--
  | [2018-02-25 20:17:42,839] INFO  <opicsWatcher:93> [oop-thread-7] Created topics: [connect-cluster-offsets]
  | [2018-02-25 20:17:44,770] INFO  <adataHandler:66> [oop-thread-7] Max attempts reached on getting metadata for connect-cluster-offsets after 1400ms, giving up for now
  | [2018-02-25 20:17:44,774] WARN  <opicsWatcher:101> [oop-thread-7] Error responding to creation of topic connect-cluster-offsets
  | io.strimzi.controller.topic.MaxAttemptsExceededException
  | at io.strimzi.controller.topic.BackOff.delay(BackOff.java:44)
  | at io.strimzi.controller.topic.BackOff.delayMs(BackOff.java:36)
  | at io.strimzi.controller.topic.TopicMetadataHandler.retry(TopicMetadataHandler.java:63)
  | at io.strimzi.controller.topic.Controller$4$1.handle(Controller.java:715)
  | at io.strimzi.controller.topic.Controller$4$1.handle(Controller.java:705)
  | at io.strimzi.controller.topic.BaseKafkaImpl.lambda$topicMetadata$0(BaseKafkaImpl.java:244)
  | at io.strimzi.controller.topic.BaseKafkaImpl$MetadataWork.complete(BaseKafkaImpl.java:186)
  | at io.strimzi.controller.topic.BaseKafkaImpl$Work.run(BaseKafkaImpl.java:54)
  | at io.strimzi.controller.topic.BaseKafkaImpl$Work.handle(BaseKafkaImpl.java:61)
  | at io.strimzi.controller.topic.BaseKafkaImpl$Work.handle(BaseKafkaImpl.java:51)
  | at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:344)
  | at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
  | at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
  | at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
  | at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
  | at java.lang.Thread.run(Thread.java:748)
  | [2018-02-25 20:17:47,266] INFO  <opicsWatcher:93> [oop-thread-1] Created topics: [__consumer_offsets]
  | [2018-02-25 20:17:47,315] WARN  <opicsWatcher:101> [oop-thread-1] Error responding to creation of topic __consumer_offsets
  | org.apache.kafka.common.errors.LeaderNotAvailableException: There is no leader for this topic-partition as we are in the middle of a leadership election.
  | [2018-02-25 20:17:50,840] INFO  <opicsWatcher:93> [oop-thread-6] Created topics: [connect-cluster-status]
```

**Comment by ppatierno** [comment]
I wasn't able to reproduce the issue maybe I was lucky or the default timeout worked fine for me.
Making it configurable means doing that for the `BackOff` class but I guess that we don't want to expose outside all three main parameters to be configurable (`scaleMs`, `base` and `maxAttempts`).
I guess that just making it configurable through `maxAttempts` could be enough. Wdyt @tombentley @scholzj  ?

**Comment by ppatierno** [comment]
@serrss it should be fixed by #311 . Can you give it a try please ?

**Comment by serrss** [comment]
@ppatierno Issue was fixed. I can't reproduce this bug after PR #311 was merged.
TC log:

```
[2018-03-15 16:43:16,444] INFO  <opicsWatcher:93> [oop-thread-2] Created topics: [connect-cluster-offsets]
[2018-03-15 16:43:22,627] INFO  <opicsWatcher:93> [oop-thread-6] Created topics: [connect-cluster-status]
[2018-03-15 16:43:23,385] INFO  <Controller  :392> [oop-thread-0] Reconciling topic connect-cluster-offsets, k8sTopic:null, kafkaTopic:nonnull, privateTopic:null
[2018-03-15 16:43:23,775] INFO  <igMapWatcher:35> [ault.svc/...] ConfigMap watch received event ADDED on map connect-cluster-offsets with labels {strimzi.io/cluster=my-cluster, strimzi.io/kind=topic}
[2018-03-15 16:43:24,419] INFO  <Controller  :392> [oop-thread-2] Reconciling topic connect-cluster-status, k8sTopic:null, kafkaTopic:nonnull, privateTopic:null
[2018-03-15 16:43:24,627] INFO  <opicsWatcher:93> [oop-thread-7] Created topics: [connect-cluster-configs]
```
